File: blockball-bukkit-api/src/main/java/com/github/shynixn/blockball/api/bukkit/event/BallKickEvent.java
package com.github.shynixn.blockball.api.bukkit.event;

import com.github.shynixn.blockball.api.entities.Ball;
import org.bukkit.entity.Player;

@SuppressWarnings("unused")
public final class BallKickEvent extends BallInteractEvent {
    public BallKickEvent(Player player, Ball ball) {
        super(player, ball);
    }
}

File: blockball-bukkit-api/src/main/java/com/github/shynixn/blockball/api/bukkit/event/BallKickEvent.kt
package com.github.shynixn.blockball.api.bukkit.event

import com.github.shynixn.blockball.api.business.proxy.BallProxy
import org.bukkit.entity.Player
import org.bukkit.util.Vector

/**
 * Event which gets called when the ball is kicked.
 */
class BallKickEvent(
    ball: BallProxy,
    player: Player,
    velocity: Vector
) : BallTouchEvent(ball, player, velocity)
